pkg/model/model.go

```diff
@@ -459,44 +459,78 @@ func (m *Model) processNestedFieldTypeDefs(
 				replaceSecretAttrGoType(crd, field, tdefs)
 			}
 			if field.FieldConfig.References != nil {
-				updateTypeDefAttributeWithReference(fieldPath, tdefs)
+				updateTypeDefAttributeWithReference(fieldPath, tdefs, crd)
 			}
 		}
 	}
 }
 
 // updateTypeDefAttributeWithReference adds a new AWSResourceReference attribute
 // for the corresponding attribute represented by fieldPath of nested field
-func updateTypeDefAttributeWithReference(fieldPath string, tdefs []*TypeDef) {
+func updateTypeDefAttributeWithReference(fieldPath string, tdefs []*TypeDef, crd *CRD) {
 	fp := ackfp.FromString(fieldPath)
 	if fp.Size() < 2 {
 		// TypeDef should only be updated with Reference fields for nested fieldPath.
 		// For non-nested fieldPath, the references are added directly to the resource
 		// Spec.
 		return
 	}
-	fieldName := fp.Pop()
-	// parent of fieldName determines the TypeDef
-	parentFieldName := fp.Pop()
+	// First part of nested reference fieldPath is the name of Spec field
```

Comment:
When you say "is the name of the Spec field", are you referring to the field *within* the `Spec` field? i.e. the "top-level field" in the Spec?

Or are you referring to `Config.PrefixConfig.SpecField`? :) Because, you removed the call to `fp.Pop()` above and therefore the "first" field in the fieldpath is actually going to be the `Spec` itself...

Reply:
It's the former. I have updated the comment to be more precise.
> // First part of nested reference fieldPath is the name of top level Spec
> // field. Ex: For 'ResourcesVpcConfig.SecurityGroupIds' fieldpath the
> // specFieldName is 'ResourcesVpcConfig'

The fieldpath passed in this method is the key of ResourceConfig.Fields, which does not include Spec prefix.